Code/GraphMol/RWMol.cpp

```diff
@@ -574,26 +576,272 @@ void RWMol::beginBatchEdit() {
   dp_delAtoms.reset(new boost::dynamic_bitset<>(getNumAtoms()));
   dp_delBonds.reset(new boost::dynamic_bitset<>(getNumBonds()));
 }
+
 void RWMol::commitBatchEdit() {
-  if (!dp_delBonds || !dp_delAtoms) {
-    return;
-  }
-  auto delBonds = *dp_delBonds;
-  dp_delBonds.reset();
-  for (unsigned int i = delBonds.size(); i > 0; --i) {
-    if (delBonds[i - 1]) {
-      const auto bnd = getBondWithIdx(i - 1);
-      CHECK_INVARIANT(bnd, "bond not found");
-      removeBond(bnd->getBeginAtomIdx(), bnd->getEndAtomIdx());
+    if (!(dp_delBonds || dp_delAtoms)) {
+        return;
     }
-  }
-  auto delAtoms = *dp_delAtoms;
-  dp_delAtoms.reset();
-  for (unsigned int i = delAtoms.size(); i > 0; --i) {
-    if (delAtoms[i - 1]) {
-      removeAtom(i - 1);
+
+    batchRemoveBonds();
+    batchRemoveAtoms();
+    
+    // remove ring info
+    dp_ringInfo->reset();
+    
+    // fix properties
+    clearComputedProps(true);
+    dp_delBonds.reset();
+    dp_delAtoms.reset();
+}
+
+void RWMol::batchRemoveBonds() {
+    if(!dp_delBonds)
+        return;
```

Comment:
```suggestion
    if(!dp_delBonds) {
        return;
    }
```

Please fix all the rest of these

Reply:
resolved